src/openms/source/ANALYSIS/ID/ConsensusIDAlgorithmPEPMatrix.cpp

```diff
@@ -42,88 +42,59 @@ namespace OpenMS
   }
 
-
   void ConsensusIDAlgorithmPEPMatrix::updateMembers_()
   {
     ConsensusIDAlgorithmSimilarity::updateMembers_();
 
-    // alignment scoring using SeqAn/similarity matrices:
-    std::string matrix = param_.getValue("matrix");
+    string matrix = param_.getValue("matrix");
     int penalty = param_.getValue("penalty");
-    scoring_method_ = SeqAnScore(-penalty, -penalty);
-    if (matrix == "identity")
-    {
-      ::seqan::setDefaultScoreMatrix(scoring_method_, 
-                                     ::seqan::AdaptedIdentity());
-    }
-    else if (matrix == "PAM30MS")
+
+    alignment_.setMatrix(matrix);
+
+    if (penalty > 0)
     {
-      ::seqan::setDefaultScoreMatrix(scoring_method_, ::seqan::PAM30MS());
+      alignment_.setPenalty(penalty);
     }
     else
     {
-      String msg = "Matrix '" + matrix + "' is not known! Valid choices are: "
-        "'identity', 'PAM30MS'.";
+      String msg = "Gap penalty should be positive";
       throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
                                        msg);
     }
-
     // new parameters may affect the similarity calculation, so clear cache:
     similarities_.clear();
   }
 
-
   double ConsensusIDAlgorithmPEPMatrix::getSimilarity_(AASequence seq1,
                                                        AASequence seq2)
   {
     // here we cannot take modifications into account:
     String unmod_seq1 = seq1.toUnmodifiedString();
     String unmod_seq2 = seq2.toUnmodifiedString();
     if (unmod_seq1 == unmod_seq2) return 1.0;
-    // order of sequences matters for cache look-up:
-    if (unmod_seq1 > unmod_seq2) swap(unmod_seq1, unmod_seq2);
-    seq1 = AASequence::fromString(unmod_seq1);
-    seq2 = AASequence::fromString(unmod_seq2);
-    pair<AASequence, AASequence> seq_pair = make_pair(seq1, seq2);
-    SimilarityCache::iterator pos = similarities_.find(seq_pair);
-    if (pos != similarities_.end()) return pos->second; // score found in cache
-    
-    // use SeqAn similarity scoring:
-    SeqAnSequence seqan_seq1 = unmod_seq1.c_str();
-    SeqAnSequence seqan_seq2 = unmod_seq2.c_str();
-    // seq. 1 against itself:
-    ::seqan::assignSource(row(alignment_, 0), seqan_seq1);
-    ::seqan::assignSource(row(alignment_, 1), seqan_seq1);
-    double score_self1 = globalAlignment(alignment_, scoring_method_,
-                                         ::seqan::NeedlemanWunsch());
-    // seq. 1 against seq. 2:
-    ::seqan::assignSource(row(alignment_, 1), seqan_seq2);
-    double score_sim = globalAlignment(alignment_, scoring_method_, 
-                                       ::seqan::NeedlemanWunsch());
-    // seq. 2 against itself:
-    ::seqan::assignSource(row(alignment_, 0), seqan_seq2);
-    double score_self2 = globalAlignment(alignment_, scoring_method_,
-                                         ::seqan::NeedlemanWunsch());
+    if (unmod_seq1 < unmod_seq2) swap(unmod_seq1, unmod_seq2);
```

Comment:
no need to do this anymore, right (because the alignment score is unaffected and the caching is gone now)